src/cpl/share_esmf/ZenderSoilErodStreamType.F90

```diff
@@ -0,0 +1,365 @@
+              stream_mapalgo      = control%zendersoilerod_mapalgo,                       &
+              stream_filenames    = (/trim(control%stream_fldFileName_zendersoilerod)/), &
+              stream_fldlistFile  = stream_varnames,                                    &
+              stream_fldListModel = stream_varnames,                                    &
+              stream_yearFirst    = 2003,                                               &
+              stream_yearLast     = 2003,                                               &
+              stream_yearAlign    = 1,                                                  &
+              stream_offset       = 0,                                                  &
+              stream_taxmode      = 'extend',                                           &
+              stream_dtlimit      = 1.0e30_r8,                                          &
+              stream_tintalgo     = 'linear',                                           &
+              stream_name         = 'Zender soil erodibility',                          &
+              rc                  = rc)
+         if (ESMF_LogFoundError(rcToCheck=rc, msg=ESMF_LOGERR_PASSTHRU, line=__LINE__, file=__FILE__)) then
+            write(iulog,*) 'Error on stream initialize -- see PET*.ESMF_LogFile(s)'
+            call endrun("ESMF log error")
+         end if
+
+         ! Explicitly set current date to a hardcoded constant value. Otherwise
+         ! using the real date can cause roundoff differences that are
+         ! detrected as issues with exact restart.  EBK M05/20/2017
+         ! call get_curr_date(year, mon, day, sec)
+         year = 2003
+         mon  = 12
+         day  = 31
+         sec  = 0
+         mcdate = year*10000 + mon*100 + day
+
+         call shr_strdata_advance(sdat_erod, ymd=mcdate, tod=sec, logunit=iulog, istr='zendersoilerod', rc=rc)   ! what is istr and do I need to change elsewhere because the change of istr here
+         if (ESMF_LogFoundError(rcToCheck=rc, msg=ESMF_LOGERR_PASSTHRU, line=__LINE__, file=__FILE__)) then
+            write(iulog,*) 'Error on stream advance -- see PET*.ESMF_LogFile(s)'
+            call endrun("ESMF log error")
+         end if
+
+         ! Get pointer for stream data that is time and spatially interpolate to model time and grid
+         do n = 1,size(stream_varnames) 
+            call dshr_fldbun_getFldPtr(sdat_erod%pstrm(1)%fldbun_model, stream_varnames(n), fldptr1=dataptr1d, rc=rc)
+            if (ESMF_LogFoundError(rcToCheck=rc, msg=ESMF_LOGERR_PASSTHRU, line=__LINE__, file=__FILE__)) then
+               write(iulog,*) 'Error on get field pointer -- see PET*.ESMF_LogFile(s)'
+               call endrun("ESMF log error")
+            end if
+            if (trim(stream_varnames(n)) == 'mbl_bsn_fct_geo') then
+               ig = 0
+               do g = bounds%begg,bounds%endg
+                  ig = ig+1                   ! not sure why +1 is needed but it's okay
```

Comment:
The +1 is needed so that the index in dataptr1d gets incremented as g is updated. This is a critical part of this code.

Reply:
Fixed